s3fs/core.py

```diff
@@ -1351,14 +1394,23 @@ async def _copy_managed(self, path1, path2, size, block=5 * 2 ** 30, **kwargs):
         )
         self.invalidate_cache(path2)
 
-    async def _cp_file(self, path1, path2, **kwargs):
-        gb5 = 5 * 2 ** 30
+    async def _cp_file(self, path1, path2, preserve_etag=None, **kwargs):
+        threshold = kwargs.pop("managed_threshold", 5 * 2 ** 30)
         path1 = self._strip_protocol(path1)
         bucket, key, vers = self.split_path(path1)
-        size = (await self._info(path1, bucket, key, version_id=vers))["size"]
-        if size <= gb5:
+
+        info = await self._info(path1, bucket, key, version_id=vers)
+        size = info["size"]
+        if size <= threshold:
             # simple copy allowed for <5GB
             await self._copy_basic(path1, path2, **kwargs)
```

Comment:
Does basic preserve the ETag?

Reply:
> Does basic preserve the ETag?

For basic it is just the md5, which is always the same.